src/context-providers/builder/types.ts

```diff
@@ -13,8 +11,10 @@ export enum EFormBuilderMode {
 // STATE
 // =============================================================================
 export type TBuilderState = {
-    mode: EFormBuilderMode;
+    mode: EBuilderMode;
     showSidePanel: boolean;
+    fields: Map<string, TField>;
+    focusedField: TField | null;
```

Comment:
instead of `null`, can we update `focusedField` to be optional, this way it can accept `undefined` 

```suggestion
    focusedField?: TField;
```